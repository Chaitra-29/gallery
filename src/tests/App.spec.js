import App from "./../App";
import { cleanup, screen,render,fireEvent } from "@testing-library/react";
import * as api from "./../util/api";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

describe("App.js", () => {
  afterEach(cleanup);
  it("renders check snapshot", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("state value present", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.instance().state).toEqual({
      checked: false,
      filter: false,
      loading: true,
      page: 1,
      photos: [],
      tags: '"mountains","dogs"',
    });
  });
  describe("fetchPhotos", () => {
    it("getPhotosfromTags", async () => {
      api.getPhotosfromTags = jest.fn();
      api.getPhotos = jest.fn();
      let wrapper = shallow(<App />);
      wrapper.instance().fetchPhotos();
      expect(api.getPhotosfromTags).toBeCalled();
    });
  });
  describe("componentDidMount", () => {
    let fetchPhotosSpy, componentDidMountSpy;
    let obj = { photos: [{ photo: "photo" }], page: 2 };
    beforeEach(() => {
      fetchPhotosSpy = jest
        .spyOn(App.prototype, "fetchPhotos")
        .mockResolvedValue(obj);
      componentDidMountSpy = jest.spyOn(App.prototype, "componentDidMount");
      shallow(<App />);
    });

    it("componentDidMount and fetchPhotosSpy are called", async () => {
      expect(fetchPhotosSpy).toBeCalledTimes(1);
      expect(componentDidMountSpy).toHaveBeenCalled();
    });
  });
});
