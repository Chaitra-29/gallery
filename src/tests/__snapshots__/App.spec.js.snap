// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js renders check snapshot 1`] = `
<div
  id="container"
>
  <Header
    onSearch={[Function]}
  />
  <Filter
    checked={false}
    onFilter={[Function]}
  />
  <Pagination
    handleNext={[Function]}
    handlePrevious={[Function]}
    loading={true}
    page={1}
  />
  <Gallery
    photos={Array []}
  />
  <Loading
    loading={true}
  />
</div>
`;
